src/Workspaces/Core/Portable/Workspace/Host/HostSolutionServices.cs

```diff
@@ -46,4 +46,10 @@ public bool IsSupported(string languageName)
         public HostProjectServices GetProjectServices(string languageName)
             => _services.GetLanguageServices(languageName).ProjectServices;
     }
+
+    internal static class HostSolutionServicesExtensions
+    {
+        public static TLanguageService GetRequiredLanguageService<TLanguageService>(this HostSolutionServices services, string language) where TLanguageService : ILanguageService
```

Comment:
>GetRequiredLanguageService [](http://example.com/codeflow?start=39&length=26)

Any reason this is not an instance method rather then extension method?

Reply:
i was currently just following our existing pattern for this. we could certainly move it to an instance method, though it will increase the size of the API review :)

Reply:
It can be an internal instance method

Reply:
it totally could be.  I'm just trying to mirror the existing patterns as i find i'm saner that way.  e.g. if the existing code i'm porting over worked by use of an extension method on something, it's easier to keep this guy an extension method in my brain so i get where it is/how it works/etc.   another way of putting it:  this is not hte interesting part of the upcoming work (IMO) so i'm doing what i can do use as little brain cycles as possible on it because there will be upcoming parts that are much harder and will need dedicate work.